src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -148,6 +135,12 @@ static bool canOptimizeListElement(BoundNode element, MethodSymbol addMethod)
                 return false;
             }
 
+            static bool usesSingleParameterBuilderMethod(CSharpCompilation compilation, BoundCollectionExpression node, TypeWithAnnotations elementType)
+            {
+                return Binder.GetCollectionBuilderMethod(node) is { Parameters: [var parameter] } &&
+                    parameter.Type.Equals(compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T).Construct([elementType]));
```

Comment:
>parameter.Type.Equals(compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T).Construct([elementType])) [](http://example.com/codeflow?start=20&length=113)

Do we have a test for the second part of the condition?

Reply:
Also, can this either be a `ReferenceEquals`, or be passed some `TypeCompareKind`?